packages/db/src/lib/execution.ts

```diff
@@ -16,6 +16,49 @@ export interface ArnRecord {
 
 const log = new Logger({ sender: '@cumulus/db/lib/execution' });
 
+/**
+ * Returns execution info sorted by most recent first for an input
+ * set of Granule Cumulus IDs.
+ *
+ * @param {Object} params
+ * @param {Knex | Knex.Transaction} params.knexOrTransaction
+ *   Knex client for reading from RDS database
+ * @param {Array<string>} params.executionColumns - Columns to return from executions table
+ * @param {number[]} params.granuleCumulusIds - The primary ID for a batch of granules
+ * @returns {Promise<Partial<PostgresExecutionRecord>[]>}
+ *   Array of arn objects with the most recent first.
+ */
+export const getExecutionInfoByGranuleCumulusIds = async ({
```

Comment:
If you're pulling records via the join table, is there any reason at all to join granules?

Reply:
yes!
because it breaks if I don't.
I did think I'd be able to say "executions corresponding to execution_cumulus_ids in granule_executions with this granule_cumulus_id"
but this is definitely something I don't understand with how postgres/sql works.
you'll see the below method I've mostly copied from is also hitting granules table for seemingly no (presumably the same) reason

Reply:
I think you just need to change the .whereIn to reference granulesExecutions.granule_cumulus_id, right?

Reply:
fixed, thanks for your help with this!